Makefile

```diff
@@ -100,23 +100,25 @@ ifdef TAG
 ENV_VARS+=-e TAG="$(TAG)"
 endif
 
-RUN = $(CONTAINER_CLI) run -t -i --sig-proxy=true -u $(UID):$(GID) --rm \
+RUN = $(CONTAINER_CLI) run --sig-proxy=true -u $(UID):$(GID) --rm \
 	-e IN_BUILD_CONTAINER="$(BUILD_WITH_CONTAINER)" \
 	-e TZ="$(TIMEZONE)" \
 	-e TARGET_ARCH="$(TARGET_ARCH)" \
 	-e TARGET_OS="$(TARGET_OS)" \
 	-e TARGET_OUT="$(TARGET_OUT)" \
 	-e TARGET_OUT_LINUX="$(TARGET_OUT_LINUX)" \
 	-e USER="${USER}" \
+	-e GOPATH="/work" \
+	-e GOBIN="/work/bin" \
 	$(ENV_VARS) \
 	-v /etc/passwd:/etc/passwd:ro \
 	$(DOCKER_SOCKET_MOUNT) \
 	$(CONTAINER_OPTIONS) \
-	--mount type=bind,source="$(PWD)",destination="/work" \
+	--mount type=bind,source="$(GOPATH)",destination="/work" \
 	--mount type=volume,source=go,destination="/go" \
 	--mount type=volume,source=gocache,destination="/gocache" \
 	$(CONDITIONAL_HOST_MOUNTS) \
-	-w /work $(IMG)
+	-w /work/src/istio.io/istio $(IMG)
```

Comment:
a couple notes on this

This file comes from the common-files repo so it shouldn't be modified there.

Moving the src to a path like this seems pretty reasonable, especially if we start mounting other local repos so you can reference them with ../api for example. Would like @sdake opinion here as well.
Note that the build tools support for this repo was merged in a few days ago only and is off by default due to a few minor issues like this one. If we don't decide to change the folder structure like this I have a patch locally I was working on that fixes the same problem

Reply:
If we don't change the path, run unit tests through "docker build pilot-test" command will fall directly.

Reply:
I understand, but there are other options to fix this as well. We should make sure this won't have other negative impact

Reply:
Okay, what's the next step. Waiting for review from @sdake.

Reply:
Yes, but it will need to be in the common-files repo. If we remove this part from the pr I'll approve all the other changes

Reply:
Okay, Thanks. I'll remove this and leave it there.